salt/_modules/azurearm_network.py

```diff
@@ -2749,3 +2749,1408 @@ def route_tables_list_all(**kwargs):
         result = {'error': str(exc)}
 
     return result
+
+
+def virtual_network_gateway_connection_create_or_update(name, resource_group, virtual_network_gateway1, 
+							connection_type, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Creates or updates a virtual network gateway connection in the specified resource group.
+
+    :param name: The name of the virtual network gateway connection to create or update. 
+
+    :param resource_group: The name of the resource group.
+
+    :param virtual_network_gateway1: The name of the virtual network gateway that will 
+	be the first endpoint of the connection.
+
+    :param connection_type: Gateway connection type. Possible values include:
+        'IPsec', 'Vnet2Vnet', and 'ExpressRoute'
+
+    A second endpoint must be passed as a keyword argument. This endpoint may be a VirtualNetworkGateway, a 
```

Comment:
add some more comments on the different scenarios you might encounter with the different types